columnar/src/dynamic_column.rs

```diff
@@ -20,6 +22,114 @@ pub enum DynamicColumn {
     Str(StrColumn),
 }
 
+impl DynamicColumn {
+    pub fn is_numerical(&self) -> bool {
+        self.is_u64() || self.is_i64() || self.is_f64()
+    }
+    pub fn is_f64(&self) -> bool {
+        matches!(self, DynamicColumn::F64(_))
+    }
+    pub fn is_u64(&self) -> bool {
+        matches!(self, DynamicColumn::U64(_))
+    }
+    pub fn is_i64(&self) -> bool {
+        matches!(self, DynamicColumn::I64(_))
+    }
```

Comment:
I'd replace those by a `.column_type()`...
That one already have a conversion to numerical_type()